src/components/content/monitor/MonitorChart.tsx

```diff
@@ -102,175 +107,168 @@ export const MonitorChart = ({
-            clearInterval(refreshOnlyLastKnownMonitorMetrics.current);
-            refreshOnlyLastKnownMonitorMetrics.current = null;
-        }
-    };
-
-    const stopFetchMonitorMetricDataTimer = () => {
-        if (refreshMonitorMetrics.current) {
-            clearInterval(refreshMonitorMetrics.current);
-            refreshMonitorMetrics.current = null;
-        }
-    };
+    useEffect(() => {
+        if (onlyLastKnownMetricQuery.isSuccess) {
+            const data: Metric[] | undefined = onlyLastKnownMetricQuery.data;
+            if (data.length > 0) {
+                getTipInfo(serviceId, false, undefined, '', '', true);
+                setOnlyLastKnownMonitorMetricsQueue((prevQueue: Metric[][] | undefined) => {
+                    let newQueue: Metric[][];
 
-    const getOnlyLastKnownMetricResponse = (selectedServiceId: string, onlyLastKnownMetric: boolean) => {
-        void MonitorService.getMetricsByServiceId(
-            selectedServiceId,
-            undefined,
-            undefined,
-            undefined,
-            undefined,
-            onlyLastKnownMetric
-        )
-            .then((rsp: Metric[]) => {
-                if (rsp.length > 0) {
-                    getTipInfo(serviceId, false, undefined, '', '', true);
-                    setOnlyLastKnownMonitorMetricsQueue((prevQueue: Metric[][] | undefined) => {
-                        let newQueue: Metric[][];
-
-                        if (prevQueue) {
-                            newQueue = [...prevQueue, rsp];
-                            if (newQueue.length > monitorMetricQueueSize) {
-                                newQueue.shift();
-                            }
-                        } else {
-                            newQueue = [rsp];
+                    if (prevQueue) {
+                        newQueue = [...prevQueue, data];
+                        if (newQueue.length > monitorMetricQueueSize) {
+                            newQueue.shift();
                         }
+                    } else {
+                        newQueue = [data];
+                    }
 
-                        return newQueue;
-                    });
-                } else {
-                    setIsAutoRefresh(false);
-                    setOnlyLastKnownMonitorMetricsQueue([]);
-                    stopFetchOnlyLastKnownMonitorMetricDataTimer();
-                    getTipInfo('', false, 'success', 'No metrics found for the selected service.', '', true);
-                }
-            })
-            .catch((error: Error) => {
+                    return newQueue;
+                });
+            } else {
                 setIsAutoRefresh(false);
                 setOnlyLastKnownMonitorMetricsQueue([]);
-                stopFetchOnlyLastKnownMonitorMetricDataTimer();
-                setTips(error);
-            });
-    };
+                getTipInfo('', false, 'success', 'No metrics found for the selected service.', '', true);
```

Comment:
in case of this status, we must keep the 'Search' button enabled. 
Also, currently when i go back and search the metrics for the same service, no request is triggered to backend. Can you please check this. 

![image](https://github.com/eclipse-xpanse/xpanse-ui/assets/54129659/e38b61a7-8ede-4451-825d-8f0d633a6e9c)